### PR TITLE
add artist back-end

### LINKED ISSUE #4: Artist backend
Create de backend function for artist in the given database
- [ ]  new branch was created
- [ ]  added dependencies 
- [ ]  updated controllers.js in api/artist folder

### PR DESCRIPTION
**Pull request by razvanbrb**
Hey Team. Can you please check  test it. I have test it but not sure if I test it correctly. Thank you!
Issue #4

### CODE REVIEW

File: api/artists/controllers.js [hunk 1]
```diff
@@ -14,14 +14,79 @@ const controllers = {
       res.json(rows)
     });
   },
-  getOne: (req, res) => { },
+  getOne: (req, res) => {
+    const sql = `select * from artist where ArtistId = ${req.params.id}`;
```
Comment (IrynaSpyrydonova) [inline api/artists/controllers.js]: const sql = `select * from artists where ArtistId = ${req.params.id}`;  you have artists table, not artist

File: api/artists/controllers.js [hunk 2]
```diff
@@ -14,14 +14,79 @@ const controllers = {
...
+      res.json(rows);
+    })
+  },
   create: (req, res) => {
     // read row data from body
+    let errors = [];
+    if (!req.body.Name) {
+      errors.push("No name of the artist specified");
+    }
+    if (!req.body.ArtistID) {
```
Comment (IrynaSpyrydonova) [inline api/artists/controllers.js]: delete this part, you dont need it:
` if (!req.body.ArtistID) {
    errors.push("No ArtistId specified");
 }`

File: api/artists/controllers.js [hunk 3]
```diff
@@ -14,14 +14,79 @@ const controllers = {
...
+      errors.push("No ArtistId specified");
+    }
+    if (errors.length) {
+      res.status(400).json({"error":errors.join(",")});
+      return;
+    }
+    const data ={
+      name: req.body.Name,
+      artistId: Number(req.body.ArtistID)
```
Comment (IrynaSpyrydonova) [inline api/artists/controllers.js]: this part from `data` also delete:
`artistId: Number(req.body.ArtistID)`;

File: api/artists/controllers.js [hunk 4]
```diff
@@ -14,14 +14,79 @@ const controllers = {
...
+    }
+    const sql = `INSERT INTO albums ( ArtistId, Name) VALUES( "${data.artistId}", "${data.name}")`;
```
Comment (IrynaSpyrydonova) [inline api/artists/controllers.js]: in the end it should look like:
`const sql = `INSERT INTO artists (Name) VALUES("${data.name}")`;`

File: api/artists/controllers.js [hunk 5]
```diff
@@ -14,14 +14,79 @@ const controllers = {
...
+        "data":data,
+        "id":this.lastID
+      });
+    });  
   },
   update: (req, res) => {
     // read row data from body
+    const data={
+      name: req.body.Name ? `'${req.body.Name}'`:null,
+      artistId: req.body.ArtistID? Number(req.body.ArtistID):null
```
Comment (IrynaSpyrydonova) [inline api/artists/controllers.js]: this part delete: 
`artistId: req.body.ArtistID? Number(req.body.ArtistID):null`

File: api/artists/controllers.js [hunk 6]
```diff
@@ -14,14 +14,79 @@ const controllers = {
...
+    }
+
+    const sql = `update artist set Name = COALESCE(${data.name}, Name), ArtistId = COALESCE(${data.artistId}, ArtistID) where ArtistId = ${req.params.id}`;
```
Comment (IrynaSpyrydonova) [inline api/artists/controllers.js]: again, here should be artists and without ArtistId(in your case it generates automatically), so should be like: 
` const sql = `update artists set Name = COALESCE(${data.name}, Name) where ArtistId = ${req.params.id}`;`

File: api/artists/controllers.js [hunk 7]
```diff
@@ -14,14 +14,79 @@ const controllers = {
...
+      }
+      res.json({
+        "message":msg,
+        "changes":this.changes
+      });
+    });
   },
-  delete: (req, res) => { }
+  delete: (req, res) => {
+    const sql = `delete from artist where ArtistId = ${req.params.id}`;
```
Comment (IrynaSpyrydonova) [inline api/artists/controllers.js]: here also change "delete from artist" into "delete from artists"

**Review by IrynaSpyrydonova — CHANGES_REQUESTED** [review]
@razvanbrb , I wrote some comments, please change and it will work

File: api/artists/controllers.js (hunk 7 above)
Comment (razvanbrb) [reply api/artists/controllers.js]: Thanks a lot @IrynaSpyrydonova. I appreciate it.

**Review by IrynaSpyrydonova — APPROVED** [review]

File: api/artists/controllers.js (hunk 6 above)
Comment (razvanbrb) [reply api/artists/controllers.js]: But I don't understand why is not saving the changes. Because I changed it. :)

Comment (IrynaSpyrydonova) [reply api/artists/controllers.js]: @razvanbrb  Why are you asking ? 🙂 I have already approved this time, everything is working

Comment (razvanbrb) [reply api/artists/controllers.js]: @IrynaSpyrydonova  Yes you are right. I have notice after I wrote comment that you approved it. Sorry.  My bad. :)

**Review by fmkarakus — APPROVED** [review]
Nice job!
Sorry for the late review.

**Pull request merged**